lib/ci.js

```diff
@@ -13,17 +13,69 @@ var opts = {
 
 var spawned;
 
-if (existsSync('package-lock.json')) {
+var packageJson = existsSync('./package.json') && require('./package.json');
+
+if (
+	packageJson
+	&& 'packageManager' in packageJson
+) {
+	var packageManager = packageJson.packageManager;
+
+	if (typeof packageManager !== 'string') {
+		console.error('Error: Invalid `packageManager` field setting ' + packageManager);
+		process.exit(1);
+	}
+
+	var parts = packageManager.split('@');
+
+	if (parts.length !== 2) {
+		console.error('Error: Invalid `packageManager` field setting' + packageManager);
+		process.exit(1);
+	}
+
+	var pm = parts[0];
```

Comment:
Why is the version is ignored when you can just pass it into `npx`?

Reply:
`packageManager` contains the `version`, and

> `yarn@3` is not available on npm

It will adapt its version automatically

And for `npm`, it's changed by your commit: [`d98f5c2` (#8)](https://github.com/privatenumber/ci/pull/8/commits/d98f5c20c97f1e6d542814ca0aec72a062893a6e#diff-d0ffa5cba3e2ffc10a6e3f0084cb72781295db237e89956fdad7847fd1d1093dR40)

Reply:
That's too bad about yarn 3 not being on npm, but I don't think we should install dependencies with yarn 2 if yarn 3 is specified. 

And I didn't change it for npm. The version was not previously passed in:
<img width="593" alt="Screen Shot 2022-07-07 at 2 18 25 AM" src="https://user-images.githubusercontent.com/1075694/177704970-64f02598-814e-432a-9a24-9f57bc345211.png">

Reply:
> And I didn't change it for npm. The version was not previously passed in:

Again, `packageManager` contains the `version`, `"packageManager": "yarn@1.22.19"` for example.

Reply:
> That's too bad about yarn 3 not being on npm, but I don't think we should install dependencies with yarn 2 if yarn 3 is specified.

We didn't, it's running `npx yarn` instead of `yarn@2` or `yarn@3`.

Reply:
My bad, I overlooked that.

It should be done for yarn as well. If yarn 3 is specified, that's an unfortunate limitation imposed by yarn for not publishing on npm.

Reply:
`yarn` will read `.yarnrc.yml` automatically, so if `yarn@v3` is used, it will still work as expected.